apps/pkeyutl.c

```diff
@@ -628,8 +661,6 @@ static EVP_PKEY_CTX *init_ctx(const char *kdfalg, int *pkeysize,
             ctx = EVP_PKEY_CTX_new(pkey, impl);
         else
             ctx = EVP_PKEY_CTX_new_from_pkey(libctx, pkey, propq);
-        if (ppkey != NULL)
-            *ppkey = pkey;
         EVP_PKEY_free(pkey);
```

Comment:
Set `pkey =  NULL` here to avoid double-free?

Reply:
Indeed this is meanwhile needed (due to the  following `goto end` in case `ctx == NULL`).
Fixed.